lxd/device/nic_routed.go

```diff
@@ -205,6 +206,36 @@ func (d *nicRouted) validateEnvironment() error {
 	return nil
 }
 
+func (d *nicRouted) checkIPAvailability(parent string) error {
+	errs := make(chan error)
+	nAddrs := 0
+
+	ipv4AddrStr, ok := d.config["ipv4.address"]
+	if ok {
+		ipv4Addrs := util.SplitNTrimSpace(ipv4AddrStr, ",", -1, true)
+		nAddrs += len(ipv4Addrs)
+
+		go arpCheckIPsAreAvailable(parent, ipv4Addrs, errs, 100*time.Millisecond)
+	}
+
+	ipv6AddrStr, ok := d.config["ipv6.address"]
+	if ok {
+		ipv6Addrs := util.SplitNTrimSpace(ipv6AddrStr, ",", -1, true)
+		nAddrs += len(ipv6Addrs)
+
+		go ndpCheckIPsAreAvailable(parent, ipv6Addrs, errs, 100*time.Millisecond)
+	}
+
+	for i := 0; i < nAddrs; i++ {
+		err := <-errs
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
If you have multiple addresses being checked and the first one fails with an error, this will return, never reading therest of the channel causing all the other goroutines to hang indefinitely.